base/show.jl

```diff
@@ -90,7 +91,7 @@ function show_default(io::IO, x::ANY)
     nf = nfields(t)
     if nf != 0 || t.size==0
         if !show_circular(io, x)
-            recur_io = IOContext(io, :SHOWN_SET => x)
+            recur_io = IOContext(io, SHOWN_SET=x, multiline=false)
```

Comment:
Since you're discussing this: I wondered whether keyword arguments couldn't be collected into an `ImmutableDict` if it's more efficient. Is there any drawback to that?